File: DSA/Ayush_Mishra_590014844/Day_13_Question_1.py
def count_words_and_longest(s):
    words = s.strip().split()
    word_count = len(words)
    longest = max(words, key=len) if words else ""
    return word_count, longest

# Example usage
s = "  Java Python C++ "
count, longest = count_words_and_longest(s)
print(f"Word count: {count}, Longest word: \"{longest}\"")

File: DSA/Ayush_Mishra_590014844/Day_13_Question_1.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

void countWordsAndLongest(const char *str) {
    int count = 0, maxLen = 0, currLen = 0;
    const char *start = NULL, *longest = NULL;

    while (*str) {
        while (*str && isspace(*str)) str++; // Skip leading spaces
        if (*str) {
            count++;
            start = str;
            currLen = 0;
            while (*str && !isspace(*str)) {
                currLen++;
                str++;
            }
            if (currLen > maxLen) {
                maxLen = currLen;
                longest = start;
            }
        }
    }

    printf("Word count: %d, Longest word: \"", count);
    for (int i = 0; i < maxLen; i++) printf("%c", longest[i]);
    printf("\"\n");
}

int main() {
    const char *text = "  Java Python C++ ";
    countWordsAndLongest(text);
    return 0;
}
